hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/InitMetaProcedure.java

```diff
@@ -78,6 +94,7 @@ private static TableDescriptor writeFsLayout(Path rootDir, Configuration conf)
     Path tableDir = CommonFSUtils.getTableDir(rootDir, TableName.META_TABLE_NAME);
     if (fs.exists(tableDir) && !fs.delete(tableDir, true)) {
       LOG.warn("Can not delete partial created meta table, continue...");
+      throw new HBaseIOException("Specified meta table already exists.");
```

Comment:
We might not need this I believe for 2 reasons:
- we should keep the original logic here for backward compatiblity,
- you throw this exception only in the case when HBase tried to delete `tableDir`, but it wasn't successful